src/gen_rules.ml

```diff
@@ -25,34 +25,40 @@ module Gen(P : Install_rules.Params) = struct
      +-----------------------------------------------------------------+ *)
 
   module Eval_modules = Ordered_set_lang.Make(struct
-      type t = (Module.t, string * Loc.t) result
-      let name = function
-        | Error (s, _) -> s
-        | Ok m -> Module.name m
+      type t = (Module.t, Module.Name.t * Loc.t) result
+      let name r = (
+        (match r with
+         | Error (s, _) -> s
+         | Ok m -> Module.name m)
+        |> Module.Name.to_string
+      )
     end)
 
-  let parse_modules ~(all_modules : Module.t String_map.t) ~buildable =
+  let parse_modules ~(all_modules : Module.t Module.Name.Map.t) ~buildable =
     let conf : Buildable.t = buildable in
-    let standard_modules = String_map.map all_modules ~f:(fun m -> Ok m) in
-    let fake_modules = ref String_map.empty in
+    let standard_modules = Module.Name.Map.map all_modules ~f:(fun m -> Ok m) in
+    let fake_modules = ref Module.Name.Map.empty in
     let parse ~loc s =
-      let s = String.capitalize s in
-      match String_map.find all_modules s with
+      let name = Module.Name.of_string s in
+      match Module.Name.Map.find all_modules name with
       | Some m -> Ok m
       | None ->
-        fake_modules := String_map.add !fake_modules s loc;
-        Error (s, loc)
+        fake_modules := Module.Name.Map.add !fake_modules name loc;
+        Error (name, loc)
     in
     let modules =
+      let standard = Module.Name.Map.to_smap standard_modules in
       Eval_modules.eval_unordered
         conf.modules
         ~parse
-        ~standard:standard_modules
+        ~standard
+      |> Module.Name.Map.of_smap
     in
     let only_present_modules modules =
-      String_map.filter_map ~f:(function
+      Module.Name.Map.filter_map ~f:(function
         | Ok m -> Some m
-        | Error (s, loc) -> Loc.fail loc "Module %s doesn't exist." s
+        | Error ((s : Module.Name.t), loc) ->
```

Comment:
Do we need to specify the type of `s`?  Isn't `Module.Name.pp` enough?

Reply:
Yeah, that shouldn't be necessary. It was probably added to make the refactoring easier. I'll get rid of it.